doc/source/basics.rst

```diff
@@ -624,6 +624,66 @@ We can also pass infinite values to define the bins:
 Function application
 --------------------
 
+There are three main cases for function, depending on what the function
+is expecting. Pandas correspondingly offers a method for each case:
+
+1. `Tablewise Function Application`_: :meth:`~DataFrame.pipe`
+2. `Row or Column-wise Function Application`_: :meth:`~DataFrame.apply`
+3. Elementwise_ function application: :meth:`~DataFrame.applymap`
+
+.. _pipe:
+
+Tablewise Function Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 0.16.2
+
+DataFrames and Series can of course just be passed into functions.
+However, if the function needs to be called in a chain, consider using the :meth:`~DataFrame.pipe` method.
+Compare the following
+
+.. code-block:: python
+
+   # f, g, and h are functions taking and returning DataFrames
+   >>> f(g(h(df), arg1=1), arg2=2, arg3=3)
+
+with the equivalent
+
+.. code-block:: python
+
+   >>> (df.pipe(h),
+          .pipe(g, arg1=1),
+          .pipe(f, arg2=2, arg3=3)
+       )
+
+Pandas encourages the second style. It flows with the rest of pandas
+methods which return DataFrames or Series and are non-mutating by
```

Comment:
I'm being pedantic, but "pure" implies no-side effects. `df.plot()` is non-mutating, but not pure since it has the side-effect of drawing a plot. I didn't want some functional programming guru to call us out :)